src/engine/TransitivePathImpl.h

```diff
@@ -0,0 +1,258 @@
+// Copyright 2024, University of Freiburg,
+// Chair of Algorithms and Data Structures.
+// Author: Johannes Herrmann (johannes.r.herrmann(at)gmail.com)
+
+#pragma once
+
+#include <utility>
+
+#include "engine/TransitivePathBase.h"
+#include "util/Exception.h"
+#include "util/Timer.h"
+
+template <typename T>
+class TransitivePathImpl : public TransitivePathBase {
+ public:
+  TransitivePathImpl(QueryExecutionContext* qec,
```

Comment:
Also add a docstring that this is the common implementation for the `HashMap` and `BinSearch` version which only differ in the type of Map that they use.